commands/members/add.js

```diff
@@ -3,46 +3,96 @@
+let Utils = require('../../lib/utils')
 
 function * run (context, heroku) {
-  // Users receive `You'll be billed monthly for teams over 5 members.`
-  // message when going over the FREE_TEAM_LIMIT
-  const FREE_TEAM_LIMIT = 5
-  let org = context.org
+  let orgInfo = yield Utils.orgInfo(context, heroku)
+  let groupName = context.org || context.flags.team
+
+  let warnMembershipLimit = function (totalMembers) {
+    // Users receive `You'll be billed monthly for teams over 5 members.`
+    if (totalMembers === 6) {
+      cli.warn("You'll be billed monthly for teams over 5 members.")
+    }
+  }
+
+  let addMemberToOrg = function * (email, role, orgName) {
+    let request = heroku.request({
+      method: 'PUT',
+      path: `/organizations/${orgName}/members`,
+      body: {email, role}
+    })
+    yield cli.action(`Adding ${cli.color.cyan(email)} to ${cli.color.magenta(orgName)} as ${cli.color.green(role)}`, request)
+  }
+
+  let inviteMemberToTeam = function * (email, role, teamName) {
+    let request = heroku.request({
+      headers: {
+        Accept: 'application/vnd.heroku+json; version=3.team-invitations'
+      },
+      method: 'PUT',
+      path: `/organizations/${teamName}/invitations`,
+      body: {email, role}
+    }).then(request => {
+      cli.action.done('email sent')
+    })
+
+    yield cli.action(`Inviting ${cli.color.cyan(email)} to ${cli.color.magenta(teamName)} as ${cli.color.green(role)}`, request)
+  }
+
   let email = context.args.email
   let role = context.flags.role
-  let features = yield heroku.get('/account/features')
-  let orgCreationFeature = features.find(function (feature) {
-    return feature.name === 'standard-org-creation'
-  })
-
-  let request = heroku.request({
-    method: 'PUT',
-    path: `/organizations/${org}/members`,
-    body: {email, role}
-  })
-  yield cli.action(`Adding ${cli.color.cyan(email)} to ${cli.color.magenta(org)} as ${cli.color.green(role)}`, request)
-
-  let members = yield heroku.get(`/organizations/${org}/members`)
-  if ((members.length === (FREE_TEAM_LIMIT + 1)) && orgCreationFeature) {
-    cli.log("You'll be billed monthly for teams over 5 members.")
+
+  if (orgInfo.type === 'team') {
+    let teamMembers = 0
+    let teamInvites = 0
+
+    teamInvites = heroku.request({
+      headers: {
+        Accept: 'application/vnd.heroku+json; version=3.team-invitations'
+      },
+      method: 'GET',
+      path: `/organizations/${groupName}/invitations`
+    })
+
+    teamMembers = heroku.get(`/organizations/${groupName}/members`)
+
+    let orgFeatures = yield heroku.get(`/organizations/${groupName}/features`)
+
+    if (orgFeatures.filter(of => of.name === 'team-invite-acceptance').length) {
+      yield inviteMemberToTeam(email, role, groupName)
+    } else {
+      yield addMemberToOrg(email, role, groupName)
+    }
+
+    Promise.all([teamInvites, teamMembers]).then(membersAndInvites => {
+      let membersCount = membersAndInvites.reduce((m, i) => {
```

Comment:
Rather than `reduce`, destructuring might make this clearer.

```
let [members, invites] = membersAndInvites;
return members.length + invites.length;